net/client.go

```diff
@@ -386,8 +383,6 @@ func (s *server) pushRecord(ctx context.Context, id thread.ID, lid peer.ID, rec
 	if err = s.ps.Publish(ctx, id, req); err != nil {
 		log.Errorf("error publishing record: %s", err)
 	}
-
-	wg.Wait()
```

Comment:
I can't remember why we waited here... probably out of an abundance of caution while developing.